pilot/lib/pilot/build_manager.rb

```diff
@@ -128,88 +105,33 @@ def should_update_build_information(options)
-          latest_build = builds.last unless latest_build
-          break unless builds.include?(latest_build)
-
-          if latest_build.valid and must_update_build_info
-            # Set the changelog and/or description if necessary
-            if should_update_build_information(options)
-              latest_build.update_build_information!(whats_new: options[:changelog], description: options[:beta_app_description], feedback_email: options[:beta_app_feedback_email])
-              UI.success "Successfully set the changelog and/or description for build"
-            end
-            must_update_build_info = false
-          end
-
-          UI.message("Waiting for iTunes Connect to finish processing the new build (#{latest_build.train_version} - #{latest_build.build_version})")
-        end
-      end
+    def distribute_build(uploaded_build, options)
+      UI.message("Distributing new build to testers: #{uploaded_build.train_version} - #{uploaded_build.build_version}")
 
-      UI.user_error!("Error receiving the newly uploaded binary, please check iTunes Connect") if latest_build.nil?
-      full_build = nil
+      uploaded_build.export_compliance.encryption_updated = false
+      uploaded_build.beta_review_info.demo_account_required = false
+      uploaded_build.submit_for_testflight_review!
 
-      while full_build.nil? || full_build.processing
-        # The build's processing state should go from true to false, and be done. But sometimes it goes true -> false ->
-        # true -> false, where the second true is transient. This causes a spurious failure. Find build by build_version
-        # and ensure it's not processing before proceeding - it had to have already been false before, to get out of the
-        # previous loop.
-        full_build = app.build_trains(platform: platform)[latest_build.train_version].builds.find do |b|
-          b.build_version == latest_build.build_version
-        end
+      if options[:distribute_external]
+        external_group = Spaceship::TestFlight::Group.default_external_group(app_id: uploaded_build.app_id)
 
-        UI.message("Waiting for iTunes Connect to finish processing the new build (#{latest_build.train_version} - #{latest_build.build_version})")
-        sleep(wait_processing_interval)
-      end
+        if external_group.nil? && options[:groups].nil?
+          UI.user_error!("You must specify at least one group using the `:groups` option to distribute externally")
+        end
 
-      if full_build && !full_build.processing && full_build.valid
-        minutes = ((Time.now - start) / 60).round
-        UI.success("Successfully finished processing the build")
-        UI.message("You can now tweet: ")
-        UI.important("iTunes Connect #iosprocessingtime #{minutes} minutes")
-        return full_build
-      else
-        UI.user_error!("Error: Seems like iTunes Connect didn't properly pre-process the binary")
+        uploaded_build.add_group!(external_group)
```

Comment:
external_group could be nil if user has no default external group, but is specifying a group?  Also, what if user is specifying a group, this code path should not be executed.